### ISSUE TITLE
3 specs failing with "ERROR ArgumentError: negative length -39 given" or similar

### ISSUE THREAD

**Issue by wincent** [jarib/celerity#20]
I'm seeing 3 specs fail due to WEBrick raising an exception and returning a 500 error:
- http://gist.github.com/449972
- http://gist.github.com/449974
- http://gist.github.com/449975

I'm also seeing a 4th spec fail, but it is probably an unrelated issue:
- http://gist.github.com/449977 

Info about my environment:
- "jruby --version" output:
  
  jruby 1.5.1 (ruby 1.8.7 patchlevel 249) (2010-06-06 f3a3480) (Java HotSpot(TM) 64-Bit Server VM 1.6.0_20) [x86_64-java]
- Mac OS X 10.6.4
- "uname -a" output:
  
  Darwin atocha.local 10.4.0 Darwin Kernel Version 10.4.0: Fri Apr 23 18:28:53 PDT 2010; root:xnu-1504.7.4~1/RELEASE_I386 i386
- standard Ruby 1.8.7
- "java -version" output:
  
  java version "1.6.0_20"
  Java(TM) SE Runtime Environment (build 1.6.0_20-b02-279-10M3065)
  Java HotSpot(TM) 64-Bit Server VM (build 16.3-b01-279, mixed mode)

Finally, here's the verbose output of a full run ("jruby -S spec -f s --backtrace spec"):
- http://gist.github.com/449966

Let me know if I can gather any more info to help.

Cheers,
Wincent

**Comment by jarib** [comment]
Thanks. Could you run

<pre>
jruby -S gem install mongrel
</pre>


and see if that solves the first three problems. The fourth is a known issue (just filed https://sourceforge.net/tracker/?func=detail&aid=3020186&group_id=47038&atid=448269) - I'll add a guard for it momentarily.

**Comment by wincent** [comment]
Yep, the first three failures go away when the mongrel gem is installed.

I still see WEBrick getting fired up at the start of the run though, which I guess is expected:

```
......[2010-06-23 17:13:26] INFO  WEBrick 1.3.1
[2010-06-23 17:13:26] INFO  ruby 1.8.7 (2010-06-06) [java]
[2010-06-23 17:13:26] INFO  WEBrick::HTTPProxyServer#start: pid=80934 port=2001
127.0.0.1 - - [23/Jun/2010:17:13:26 CEST] "GET http://0.0.0.0:2000/ HTTP/1.1" 200 17
- -> http://0.0.0.0:2000/
........Browser#html called on big content page
F..............[2010-06-23 17:13:28] INFO  going to shutdown ...
[2010-06-23 17:13:28] INFO  WEBrick::HTTPProxyServer#start done.
...............................
```

**Comment by jarib** [comment]
Yes, we're using WEBrick to test the proxy support. Shouldn't give you any issues.

**Comment by jarib** [comment]
Add mongrel as development dependency. Closed by 6815a25bddf1161e42013286c4095a6b439649e8
